dipper/models/Dataset.py

```diff
@@ -43,163 +181,388 @@ def __init__(
         else:
-            LOG.debug('No rights provided.')
+            self.graph.addTriple(self.distribution_level_turtle_curie,
+                                 'dcterms:license',
+                                 self.license_url)
 
-        if ingest_desc is not None:
-            self.model.addDescription(self.identifier, ingest_desc)
-        return
+        if self.data_rights is not None:
+            self.graph.addTriple(self.distribution_level_turtle_curie,
+                                 'dcterms:rights',
+                                 self.data_rights)
 
-    def setVersion(self, date_issued, version_id=None):
+    def compute_triples_statistics(self, target_graph):
         """
-        Legacy function...
-            should use the other set_* for version and date
+        This method computes counts of triples in a given targetGraph and writes these
+        counts into the dataset graph (i.e. self.graph)
 
-        as of 2016-10-20  used in:
+        :param target_graph: an RDF type graph containing triples to be counted
+        :return: None
+        """
 
-        dipper/sources/HPOAnnotations.py 139:
-        dipper/sources/CTD.py             99:
-        dipper/sources/BioGrid.py        100:
-        dipper/sources/MGI.py            255:
-        dipper/sources/EOM.py             93:
-        dipper/sources/Coriell.py        200:
-        dipper/sources/MMRRC.py           77:
+        # add statistics
+        self._compute_triples_count(target_graph)
+        self._compute_distinct_subjects_count(target_graph)
+        self._compute_distinct_objects_count(target_graph)
+        self._compute_distinct_entities_count(target_graph)
+        self._compute_distinct_properties_count(target_graph)
+        self._compute_distinct_classes_count(target_graph)
+        # calculate counts of each biolink:category:
+        self._compute_indiv_class_counts(target_graph,
+                                         partition_label_base="biolink_category_counts",
+                                         predicate_iri=
+                                         "http://w3id.org/biolink/vocab/category")
+        # calculate counts of each rdf:type:
+        self._compute_indiv_class_counts(target_graph,
+                                         partition_label_base="rdf_type_counts",
+                                         predicate_iri=
+                                         "http://www.w3.org/1999/02/22-rdf-syntax-ns#type")
+
+    def _compute_triples_count(self, target_graph):
+        triples_count = len(list(target_graph.triples((None, None, None))))
+        self.graph.addTriple(self.distribution_level_turtle_curie,
+                             'void:triples',
+                             Literal(triples_count, datatype=XSD.integer))
+
+    def _compute_distinct_subjects_count(self, target_graph):
+        distinct_subjects_q = target_graph.query(
+            """SELECT(COUNT(DISTINCT ?s) as ?DistinctSubjects)
```

Comment:
@justaddcoffee do you have a sense for how much time this adds to the ingest?  Curious how this affects some of our larger sources.  If it's slow, we should consider decoupling this from the ingest and post processing.

Reply:
@kshefchek Yep, this did occur to me while I was writing this, and @TomConlin and I discussed this yesterday. It adds at least several minutes to ingests that are the size of say ZFIN. 

Tom and I decided a solution would be to add a skip_stats flag to dipper-etl.py, so that we can skip the painful sparql for ingests where time is of the essence. I did this with [this commit](https://github.com/monarch-initiative/dipper/pull/809/commits/5dd69cbf68adcc15d2d41d02b2c3235778cc4a01).

We also discussed possibly swapping out rdflib for something a little faster (redland maybe?). That library doesn't seem to be under active development though, which might cause problems down the road.

We also could postprocess, like you said, but this makes testing of the code a little more awkward, and also makes the pipeline a little less coherent IMHO. Thoughts? Glad to discuss this more if you want.

Reply:
ah sorry, I saw that you were already addressing it.  I don't have a great answer, I still think we should just use neo4j, but if that's not an option we could use redland -- or rdflib as a last resort.  I've had good luck with the redland utils even though the project seems dead.

Reply:
also - ntriples | cut | sort -u
if we're just counting things